contracts/tokenomics/maker/src/contract.rs

```diff
@@ -463,16 +464,19 @@ fn distribute_astro(deps: DepsMut, env: Env, info: MessageInfo) -> Result<Respon
     }
 
     let mut cfg = CONFIG.load(deps.storage)?;
-    let distribute_msg = distribute(deps, env, &mut cfg)?;
+    let (distribute_msg, attributes) = distribute(deps, env, &mut cfg)?;
     if distribute_msg.is_empty() {
         return Ok(Response::default());
     }
 
     Ok(Response::default()
         .add_submessages(distribute_msg)
-        .add_attribute("action", "distribute_astro"))
+        .add_attribute("action", "distribute_astro")
```

Comment:
It is better to add this attribute to distribute function, otherwise it isn't emitted when distribute function is called directly from collect function

Reply:
Good suggestion. Fixed